Jetty Maven Plugin - jetty:run does not work with pom type dependencies

Upgrading from Jetty 9.4 to 12

**Jetty version(s)**
Jetty 12

**Jetty Environment**
ee10

**Java version/vendor** `(use: java -version)`
OpenJDK 64-Bit Server VM (build 17.0.12+7-Ubuntu-1ubuntu222.04, mixed mode, sharing)
Apache Maven 3.6.3

**OS type/version**
Ubuntu 22

**Description**
This is working with Jetty 9.4.  We have a maven dependency in the web-app that is `<type>pom</type>`, more details [here](https://books.sonatype.com/mvnref-book/reference/pom-relationships-sect-pom-best-practice.html#pom-relationships-sect-grouping-deps).  When running with `jetty:run`, we get the following error.

<!-- Do not report security issues here! See [Jetty Security Reports](https://eclipse.dev/jetty/security_reports.php) -->
```
[INFO] jetty-12.0.12; built: 2024-07-25T21:58:37.668Z; git: cc6f1b74db755fed228b50701ad967aeaa68e83f; jvm 17.0.12+7-Ubuntu-1ubuntu222.04
[WARNING] Invalid Resource Reference: jar:file:/home/jay/.m2/repository/org/example/ext-maven-deps/1.0-SNAPSHOT/ext-maven-deps-1.0-SNAPSHOT.pom!/
[WARNING] Failed startup of context oeje10mp.MavenWebAppContext@6909f6be{ROOT,/,b=file:///home/jay/projects/temp/jetty-maven-proj/webapp/src/main/webapp/,a=STOPPED,h=oeje10s.SessionHandler@6b5ab2f2{STOPPED}}{file:///home/jay/projects/temp/jetty-maven-proj/webapp/src/main/webapp/}
java.lang.IllegalArgumentException: Unable to mount FileSystem from unsupported URI: jar:file:///home/jay/.m2/repository/org/example/ext-maven-deps/1.0-SNAPSHOT/ext-maven-deps-1.0-SNAPSHOT.pom!/
    at org.eclipse.jetty.util.resource.FileSystemPool.mount (FileSystemPool.java:135)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$CompositeResourceFactory.mountIfNeeded (ResourceFactoryInternals.java:268)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$CompositeResourceFactory.newResource (ResourceFactoryInternals.java:225)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$Closeable.newResource (ResourceFactoryInternals.java:145)
    at org.eclipse.jetty.util.resource.ResourceFactory.newResource (ResourceFactory.java:331)
...
Caused by: java.nio.file.ProviderNotFoundException: Provider "jar" not found
    at java.nio.file.FileSystems.newFileSystem (FileSystems.java:359)
    at java.nio.file.FileSystems.newFileSystem (FileSystems.java:288)
    at org.eclipse.jetty.util.resource.FileSystemPool.mount (FileSystemPool.java:116)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$CompositeResourceFactory.mountIfNeeded (ResourceFactoryInternals.java:268)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$CompositeResourceFactory.newResource (ResourceFactoryInternals.java:225)
    at org.eclipse.jetty.util.resource.ResourceFactoryInternals$Closeable.newResource (ResourceFactoryInternals.java:145)
    at org.eclipse.jetty.util.resource.ResourceFactory.newResource (ResourceFactory.java:331)

```
**How to reproduce?**
Extract the sample project.  Run `mvn install`, then `cd webapp`, and `mvn jetty:run`

[jetty-maven-proj.zip](https://github.com/user-attachments/files/16660271/jetty-maven-proj.zip)

--- a/jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/AbstractUnassembledWebAppMojo.java
+++ b/jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/AbstractUnassembledWebAppMojo.java
@@ -242,6 +242,10 @@ public abstract class AbstractUnassembledWebAppMojo extends AbstractWebAppMojo
         if ("war".equalsIgnoreCase(artifact.getType()))
             return false;
 
+        //The dependency cannot be a pom
+        if ("pom".equalsIgnoreCase(artifact.getType()))
+            return false;
+
         //The dependency cannot be scope provided (those should be added to the plugin classpath)
         if (Artifact.SCOPE_PROVIDED.equals(artifact.getScope()))
             return false;

